yellowbrick/features/histogram.py

```diff
@@ -0,0 +1,191 @@
+        self.ratio = ratio
+        self.space = space
+
+    # decide how many bins do you want for your data
+    def get_vline_value(self, X):
+        """
+        Gets the values to draw vertical lines on histogram
+        to help balanced binning based on number of bins the
+        user wants
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x 1
+            A matrix of n instances with 1 feature
+        bin: number of bins the user wants, default=4
+        """
+        # get the number of samples
+        samples, = X.shape
+
+        # get the index for retrieving values for the bins
+        index = int(round(samples/self.bins))
+
+        # sort the samples from low to high
+        sorted_X = np.sort(X)
+
+        # a list to store the values for the binning reference
+        values = []
+
+        # get the values for the binning reference
+        for i in range(0, (self.bins-1)):
+            value = sorted_X[index * (i + 1)]
+            values.append(value)
+
+        return values
+
+    def draw(self, X, **kwargs):
+        """
+        Draws a histogram with the reference value for binning as vetical lines
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x 1
+            A matrix of n instances with 1 feature
+        """
+
+        # draw the histogram
+        self.ax.hist(X, bins=50, color="#6897bb", **kwargs)
+
+        # add vetical line with binning reference values
+        self.ax.vlines(self.get_vline_value(X), 0, 1, transform=self.ax.get_xaxis_transform(), colors='r')
+        print("The binning reference values are:", self.get_vline_value(X))
+
+    def fit(self, X, **kwargs):
+        """
+        Sets up X for the histogram and checks to
+        ensure that X is of the correct data type
+        Fit calls draw
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x 1
+            A matrix of n instances with 1 feature
+        kwargs: dict
+            keyword arguments passed to Scikit-Learn API.
+        """
+
+        #throw an error if X has more than 1 column
+        if is_dataframe(X):
+            nrows, ncols = X.shape
+
+            if ncols > 1:
+                raise YellowbrickValueError((
+                    "X needs to be an ndarray or DataFrame with one feature, "
+                    "please select one feature from the DataFrame"
+                ))
+
+        # Handle the feature name if it is None.
+        if self.feature is None:
+
+            # If X is a data frame, get the columns off it.
+            if is_dataframe(X):
+                self.feature = X.columns
+
+            else:
+                self.feature = ['x']
+
+        self.draw(X)
+        return self
+
+
+    def poof(self, **kwargs):
+        """
+        Creates the labels for the feature and target variables
+        """
+
+        self.ax.set_xlabel(self.feature)
+        self.ax.set_ylabel(self.target)
```

Comment:
Generally speaking, we don't override `poof` since the superclass does work for showing the visualization in notebooks or in a script, my advice is to add the label setting to the finalize method.